lib/labels/cloud.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package labels
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/cloud"
+)
+
+const (
+	// AWSNamespace is used as the namespace prefix for any labels
+	// imported from AWS.
+	AWSNamespace = "aws"
+	// AzureNamespace is used as the namespace prefix for any labels
+	// imported from Azure.
+	AzureNamespace = "azure"
+	// labelUpdatePeriod is the period for updating cloud labels.
+	labelUpdatePeriod = time.Hour
+)
+
+const (
+	awsErrorMessage   = "Could not fetch EC2 instance's tags, please ensure 'allow instance tags in metadata' is enabled on the instance."
+	azureErrorMessage = "Could not fetch Azure instance's tags."
+)
+
+// CloudConfig is the configuration for a cloud label service.
+type CloudConfig struct {
+	Client               cloud.InstanceMetadata
+	Clock                clockwork.Clock
+	Log                  logrus.FieldLogger
+	namespace            string
+	instanceMetadataHint string
+}
+
+func (conf *CloudConfig) checkAndSetDefaults() error {
+	if conf.Client == nil {
+		return trace.BadParameter("Missing parameter: Client")
+	}
+	if conf.Clock == nil {
+		conf.Clock = clockwork.NewRealClock()
+	}
+	if conf.Log == nil {
+		conf.Log = logrus.WithField(trace.Component, "ec2labels")
+	}
+	return nil
+}
+
+// CloudImporter is a service that periodically imports tags from a cloud service via instance
+// metadata.
+type CloudImporter struct {
+	*CloudConfig
+	mu     sync.RWMutex
```

Comment:
Can you rename the mutex to `muLabels` so the name indicates that it protects the `labels`?